Added a settings page.

techfreek opened the pull request:
No more XML editing! I also attempted to clean up a few areas.

Here is what it looks like in game:
![settingspage](https://cloud.githubusercontent.com/assets/3945675/9753361/87205b72-5670-11e5-80f1-14b1ecee03b4.PNG)

techfreek committed:
Added a settings page.

No more XML editing! I also attempted to clean up a few areas.

enkafan commented:
this is great. I'll pull it down tonight and check it out and try and get it released. ignore the build failure, I should have killed the appveyour project for this a while ago

techfreek commented:
I need to fix our thing before you merge. I just realized that even if you don't click save that the changes will still apply in game....  They just won't be saved to disk.

techfreek committed:
Added staged changes, updating changes in the ColorMonitor thread, and
improved the logger.

Changes in the settings window are no longer directly updated in game.
Well, they weren't able to update without a restart until now. This allows
changes to be seen in game. The console is now a singleton so it is easier
to write to a log file. Which is really nice when something happens during
map load and you can't see the console.

techfreek committed:
Style and debug.

Cleaned up some code and added a way to enable debug from the settings
panel. That is off by default.

techfreek commented:
Okay, should be ready now. Let me know if I need to fix anything

techfreek commented:
How does this look? Do I need to fix anything?

enkafan commented:
hey, real life got in the way. I'm going to merge this in now but I suspect that After Dark might cause some overall breaking change with some of the new transport stuff. Not sure if releasing now makes sense or waiting until after AD is released.

techfreek commented:
It's all good. I know how that can be. You make a good point about After Dark. Many things could break. The only reason why it might be smart to merge now is because of the improvements to logging I made.  

Maybe instead of merging this directly to master, maybe you should create an AfterDark branch. This can be pulled into that so the logging improvements can be used to fix any bugs that come up. Once that branch is stable, it can be merged to master and released.

enkafan commented:
Good call. Just created that branch with these changes. Hopefully with the new release things will go smoothly and it'll be a nice new polished mod for everyone revisiting the game with the new content.

I don't think there will be breaking changes, but based on how things like stations were included as a "route" make me worry about some of the new transport options. Hopefully they kept it simple, but you never know.

I'll leave this open and let it close automatically once we merge the after-dark branch into master

techfreek commented:
Awesome. If you find anything in that branch that needs to be fixed, let me know. Hopefully the amount of breaking changes is small.

enkafan commented:
well, good news and bad news. Good news is this PR fixed a null reference exception that was thrown. So I went ahead and published it to keep the game from crashing. The bad news is it looks it is detecting lines and applying colors but they aren't sticking. I'm at work right now so my debugging time is limited until tonight

techfreek committed:
Logging, cleanup, and after dark

Added some more logging to  the color manager, cleaned up some of the
code. I tweaked the main loop so we don't have to loop through every line
if we are not using it. And tried to do some after dark fixes, but had no
luck what so ever.

techfreek committed:
Got colors to save, not names.

Managed to get colors to update, but no matter what I tried, I couldn't
get naming to work.
